lighthouse-core/lib/dependency-graph/node.js

```diff
@@ -5,6 +5,12 @@
  */
 'use strict';
 
+/**
+ * A union of all types derived from Node, allowing type check disrimination
+ * based on `node.type`. If a new node type is created, it should be added here.
+ * @typedef {import('./cpu-node.js') | import('./network-node.js')} NodeType
```

Comment:
how would you feel about calling this one `Node` and the real base class `BaseNode` or something? it feels weird to say that all those methods accept a thing of type `NodeType` instead of saying they expect a `Node`